### DISCUSSION TITLE
Are there any know problems using IKVM.Maven.Sdk 1.6.1 with .NET framework 4.8?

### DISCUSSION THREAD

**Discussion by martin-honnen** [ikvmnet/ikvm#443] [Q&A]
I am encountering some problems to use the latest IKVM.Maven.Sdk 1.6.1 in a .NET framework 4.8 console application.

The app seems to build but on running fails with e.g.

```
Unbehandelte Ausnahme: System.TypeInitializationException: Der Typeninitialisierer für "java.lang.Object" hat eine Ausnahme verursacht. ---> System.TypeInitializationException: Der Typeninitialisierer für "java.io.FileInputStream" hat eine Ausnahme verursacht. ---> System.TypeInitializationException: Der Typeninitialisierer für "IKVM.Runtime.LibJava" hat eine Ausnahme verursacht. ---> System.TypeInitializationException: Der Typeninitialisierer für "IKVM.Runtime.LibJvm" hat eine Ausnahme verursacht. ---> IKVM.Runtime.InternalException: Could not load libjvm.
   bei IKVM.Runtime.LibJvm..ctor()
   bei IKVM.Runtime.LibJvm..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei IKVM.Runtime.LibJava..ctor()
   bei IKVM.Runtime.LibJava..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei IKVM.Runtime.BootstrapClassLoader..ctor(RuntimeContext context)
   bei IKVM.Runtime.RuntimeClassLoaderFactory.GetBootstrapClassLoader()
   bei IKVM.Runtime.RuntimeClassLoaderFactory.GetClassLoaderWrapper(ClassLoader javaClassLoader)
   bei IKVM.Runtime.JNI.JNIFrame.GetFuncPtr(CallerID callerID, String clazz, String name, String sig)
   bei java.io.FileInputStream.initIDs()
   bei java.io.FileInputStream..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei java.io.FileInputStream.__<clinit>()
   bei java.lang.System.initializeSystemClass()
   bei IKVM.Runtime.JVM.EnsureInitialized()
   bei java.lang.Object..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei java.lang.Object..ctor()
   bei net.sf.saxon.s9api.Processor..ctor(Boolean licensedEdition) in Processor.java:Zeile 67.
   bei SaxonHE11IKVM871Net6Net48ConsoleAppTest1.Program.Main(String[] args) in C:\Users\marti\source\repos\SaxonHE11IKVM871Net6Net48ConsoleAppTest1\Program.cs:Zeile 12.			

```

The same code runs fine under .NET 6.

A sample project targeting both net6.0 and net48 is at https://github.com/martin-honnen/SaxonHE11IKVM871Net6Net48ConsoleAppTest1; as I said, it runs fine under .NET 6 and Windows 11 but fails to run with .NET framework 4.8 and Windows 11.

**Comment by wasabii** [comment]
I have not yet seen this particular bug, no. It looks like, somehow, the build process is thinking it is supposed to be generating a x86 assembly. And thus it only outputs ikvm/win-x86. However, at runtime, it's x64. Not sure yet why.

You can probably explicitely set RuntimeIdentifiers to win-x64 and it'll work.

**Reply by wasabii** [reply]
https://github.com/dotnet/sdk/issues/1545

Fun.

Okay, so the issue is that when RuntimeIdentifer isn't set, when Platform is AnyCPU, and when PlatformTarget isn't set, these are auto-infered to be win7-x86. And thus IKVM decides to copy the win-x86 native libraries.

However, it also sets Prefer32Bit to false. Which means the produced application CAN RUN ON 64 if on a 64 bit OS. Which it is, of course. So it does.

So, I think what I'll do iss add a condition specifically for Windows in here. If the RuntimeIdentifier is a win-x86 match, but Prefer32Bit is not true, we'll also copy win-x86 assets.

**Comment by martin-honnen** [comment]
I guess https://github.com/ikvmnet/ikvm/issues/444 is about resolving this. Let me know whether I should close this discussion thread.

**Reply by wasabii** [reply]
Yup
